core/lib/instrument_api.h

```diff
@@ -2780,6 +2780,20 @@ DR_API
 void
 dr_recurlock_lock(void *reclock);
 
+DR_API
+/**
+ * Acquires \p reclock, or increments the ownership count if already owned.
+ * Calls to this method which block (i.e. when the lock is already held) are
+ * marked safe to suspend AND transfer; in that case the provided mcontext will
+ * overwrite the current thread's mcontext. The provided mcontext must have a
+ * valid PC field.
+ *
+ * Callers of this function must resume application execution via
+ * dr_redirect_execution, lest they return into a flushed code page.
+ */
```

Comment:
Please add xrefs from the mark_as_app_lock routine and the mark_safe_to_suspend routine docs.  Please augment the as_app_lock to talk about it not being sufficient on its own.  Probably this routine here should xref to safe_to_suspend and highlight the difference too.

Reply:
Done